Tools/TestWebKitAPI/Tests/WebKitCocoa/OpenAndCloseWindow.mm

```diff
@@ -178,6 +184,92 @@ - (void)_webView:(WKWebView *)webView createWebViewWithConfiguration:(WKWebViewC
     TestWebKitAPI::Util::run(&isDone);
 }
 
+@interface OpenWindowUIDelegate : NSObject <WKUIDelegate>
+@end
+
+@implementation OpenWindowUIDelegate
+
+- (WKWebView *)webView:(WKWebView *)webView createWebViewWithConfiguration:(WKWebViewConfiguration *)configuration forNavigationAction:(WKNavigationAction *)navigationAction windowFeatures:(WKWindowFeatures *)windowFeatures
+{
+    openedWebView = adoptNS([[WKWebView alloc] initWithFrame:NSMakeRect(0, 0, 800, 600) configuration:configuration]);
+    if (navigationAction && navigationAction._isUserInitiated)
+        wasUserInitiated = true;
+    if (navigationAction && navigationAction._userInitiatedAction && !navigationAction._userInitiatedAction.consumed)
+        hadUnconsumedUserGesture = true;
+
+    isDone = true;
+    return openedWebView.get();
+}
+
+@end
+
+TEST(WebKit, OpenAndCloseWindowNoGestureGesture)
+{
+    resetToConsistentState();
+
+    auto delegate = adoptNS([[OpenWindowUIDelegate alloc] init]);
+
+    auto webView = adoptNS([TestWKWebView new]);
+    [webView setUIDelegate:delegate.get()];
+
+    [webView configuration].preferences.javaScriptCanOpenWindowsAutomatically = YES;
+    [webView configuration].preferences._verifyWindowOpenUserGestureFromUIProcess = YES;
+
+    NSURLRequest *request = [NSURLRequest requestWithURL:[[NSBundle mainBundle] URLForResource:@"open-window-button" withExtension:@"html" subdirectory:@"TestWebKitAPI.resources"]];
+    [webView synchronouslyLoadRequest:request];
```

Comment:
You could call `synchronouslyLoadTestPageNamed:@"open-window-button"]`. Same comment likely applies to other tests.